services/blockchain-indexer/shared/utils/snapshot.js

```diff
@@ -0,0 +1,184 @@
+};
+
+const calculateSHA256 = async (file) => new Promise((resolve, reject) => {
+	const hash = crypto.createHash('sha256');
+	const stream = fs.createReadStream(file);
+
+	stream.on('data', (data) => hash.update(data));
+	stream.on('end', () => resolve(hash.digest('hex')));
+	stream.on('error', (error) => reject(error));
+});
+
+const downloadUnzipAndVerifyChecksum = async (fileUrl, filePath) => {
+	const checksumUrl = fileUrl.replace('.gz', '.SHA256');
+
+	return new Promise((resolve, reject) => {
+		// Download the checksum file
+		logger.info('Attempting to download the snapshot checksum.');
+		getHTTPProtocolByURL(checksumUrl).get(checksumUrl, (response) => {
+			if (response.statusCode === 200) {
+				let checksumData = '';
+
+				response.on('data', (chunk) => {
+					checksumData += chunk;
+				});
+
+				response.on('end', () => {
+					// Extract the SHA256 hash from the downloaded data
+					const checksum = checksumData.trim().split(' ')[0];
+
+					// Download and unzip the file
+					logger.info('Attempting to download the snapshot file.');
+					getHTTPProtocolByURL(fileUrl).get(fileUrl, (res) => {
+						if (res.statusCode === 200) {
+							const unzip = zlib.createUnzip();
+							const writeFile = fs.createWriteStream(filePath);
+
+							res.pipe(unzip).pipe(writeFile);
+
+							res.on('error', (err) => {
+								reject(new Error(err));
+							});
+
+							writeFile.on('finish', () => {
+								// calculate hash from file.
+								calculateSHA256(filePath).then((calculatedChecksum) => {
+									if (calculatedChecksum === checksum) {
+										resolve();
+									} else {
+										reject(new Error('Checksum verification failed.'));
+									}
+								}).catch(err => {
+									reject(err);
+								});
+							});
+
+							writeFile.on('error', (err) => {
+								reject(err);
+							});
+						} else {
+							const errMessage = `Download failed with HTTP status code: ${res.statusCode} (${res.statusMessage})`;
+							console.error(errMessage);
+							if (res.statusCode === 404) {
+								reject(new Error(`NotFoundException: ${errMessage}`));
+							} else {
+								reject(new Error(errMessage));
+							}
+						}
+					}).on('error', (err) => {
+						reject(new Error(err));
+					});
+				});
+			} else {
+				reject(new Error('Failed to download checksum file.'));
+			}
+		}).on('error', (err) => {
+			reject(new Error(err));
+		});
+	});
+};
+
+const resolveSnapshotRestoreCommand = async (connEndpoint) => {
+	await checkCommandAvailability();
+	const [user, password] = connEndpoint.split('//')[1].split('@')[0].split(':');
+	const [host, port, database] = connEndpoint.split('@')[1].split(new RegExp('/|:', 'g'));
+	const mysqlSnapshotCommand = `mysql ${database} -h ${host} -P ${port} -u ${user} -p${password} < ${snapshotFilePath}`;
+	return mysqlSnapshotCommand;
+};
+
+const applySnapshot = async (connEndpoint = config.endpoints.mysql) => {
```

Comment:
Could you please define a constant at the top similar to other files and assign that to be the default value?